pkg/cloudproviders/k8s/k8s.go

```diff
@@ -0,0 +1,380 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"k8s.io/client-go/util/flowcontrol"
+
+	"github.com/ash2k/stager/wait"
+
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/atlassian/gostatsd"
+	"github.com/atlassian/gostatsd/pkg/util"
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+
+	core_v1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	core_v1inf "k8s.io/client-go/informers/core/v1"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+var (
+	// DefaultAnnotationTagWhitelist is the default ParamAnnotationTagWhitelist. Everything beginning with a standard
+	// gostatsd string is included by default.
+	DefaultAnnotationTagWhitelist = []string{
+		fmt.Sprintf("^%s", AnnotationPrefix),
+	}
+	// DefaultLabelTagWhitelist is the default ParamLabelTagWhitelist. Every label is ignored by default.
+	DefaultLabelTagWhitelist []string
```

Comment:
I think we should start making mutable things private or removing them completely (i.e. replace with a method that returns things).

Reply:
I'm all for encapsulation, I was just following the existing standard. Separate PR to encapsulate?

Reply:
Up to you. Separate PR is totally fine. 👍